### PR TITLE
Update sized vectors dependency

### PR DESCRIPTION
**Pull request by m-bock**
- Added a test that makes the existing `findMaxIndex` pass.
- Updated `sized-vectors` dependency
- Changed the existing `findMaxIndex` to work with latest `sized-vectors` package.

### CODE REVIEW

**Comment by csicar** [comment]
Thank you for the contribution. The changes definitely make the code simpler.

It would be great to also have a base case test `findMaxIndex` even if just as documentation :)

I'll merge it for now, as I see that you have another PR waiting for this to be resolved

**Comment by m-bock** [comment]
I was about to do this but there's a `Pos` constraint in the signature. So the base case of an empty vector (i guess that's what you mean) is not possible.

Generally I think it's better to use `fromRight` and `unsafePartial` in this case. As it makes clear that this case can never happen. 

Or id I get something wrong?

File: test/Main.purs [hunk 1]
```diff
@@ -224,3 +231,20 @@ main = runTest do
         sa = resize a
         sa' = matrix22 1.0 4.0 3.0 0.0
       equal sa' sa
+
+    test "findMaxIndex" do
```
Comment (csicar) [inline test/Main.purs:235]: Yeah, I didn't communicate that well. What I meant with "base case" is a case, that the test case does not test a property, but instead a particular value. 
I find it useful to have some concrete test cases as they can serve as examples on how to use the library.

Adding such a test here might also be helpful. 
Sth. like `equal 10 $ findMaxIndex (1 +> 10 +> 3 +> Vec.empty)`

Comment (m-bock) [reply test/Main.purs:235]: ok, now I understand. Sure I'll add one..

**Pull request merged**